src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -2303,6 +2303,53 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
         tmp = nullptr;
     }
 
+    void make_type_conversions(Allocator &al, std::string call_name, Vec<ASR::call_arg_t> args,
+                                const Location &loc) {
```

Comment:
Given that there is no common code for int/float, let's create two functions:

* `handle_intrinsic_int`
* `handle_intrinsic_float`

And in the caller call the appropriate one.